### PR TITLE
fixing submit button

### PR DESCRIPTION
**Pull request by utas-tajona**

### CODE REVIEW

**Comment by utas-tajona** [comment]
https://github.com/aodn/backlog/issues/2890

**Review by jonescc — CHANGES_REQUESTED** [review]
I get 'Request method 'POST' not supported' when trying to submit - needs those changes you made in the other PR that's being held up by the ui_species_attributes view?

File: ui/src/components/import/reducers/create-import.js [hunk 1]
```diff
@@ -122,9 +113,9 @@ const importSlice = createSlice({
           acc[err.id] = err.errors;
           return acc;
         }, {});
-
-        const validationErrors = mergeErrors(action.payload.errors);
-        state.enableSubmit = validationErrors.filter((err) => err.level === 'BLOCKING').length === 0;
+       const errorsList =  action.payload.errors.map(err =>err.errors);
+        const validationErrors = errorsList.reduce((acc,err) => [...acc, ...err], []);
+        state.enableSubmit = validationErrors.filter((err) => err.errorLevel === 'BLOCKING').length === 0;
```
Comment (nspool) [inline ui/src/components/import/reducers/create-import.js:118]: minor style point: can use `some` instead of `filter`

**Review by nspool — APPROVED** [review]

**Comment by jonescc** [comment]
> I get 'WARN 85224 --- [nio-8080-exec-3] o.s.web.servlet.PageNotFound Request method 'POST' not supported' when trying to submit

Might be related to the fact that this branch is behind master quite a bit now.  This ones based on 0.1.41 whereas edge is now 0.156  - perhaps merge in latest changes

**Comment by jonescc** [comment]
Submitting returns 'Job with given id has not been validated: 11' even though I just validated the job and no errors were returned.

**Comment by ccmoloney** [comment]
> Submitting returns 'Job with given id has not been validated: 11' even though I just validated the job and no errors were returned.

Is this still an issue @utas-tajona ?

**Comment by nspool** [comment]
@utas-tajona when this is ready for review again could you please run up a stack as well?

**Comment by nspool** [comment]
Testing against the `nrmn_dev` database is giving this error on sheet upload:
`Caused by: java.sql.BatchUpdateException: Batch entry 0 insert into nrmn.staged_job (created, sec_user_id, is_extended_size, last_updated, program_id, reference, source, status, survey_ids, id) values ('2021-04-21 21:51:06.86+00', 123456, 'FALSE', '2021-04-21 21:51:06.86+00', 2, 'testfile4UIMaria2019.xlsx', 'INGEST', 'PENDING', NULL, 2) was aborted: ERROR: column "survey_ids" of relation "staged_job" does not exist`

**Comment by nspool** [comment]
After adding the `survey_ids` column, noticing that Validate needs to be pressed twice after an edit for it to have an effect. Not sure if this is a new issue or a regression.

**Comment by utas-tajona** [comment]
You can run  with ddl_auto=update or you can use edge db.
________________________________

**Comment by nspool** [comment]
Also running into an Internal Server Error. Getting the same error even when no sheets are staged and with a new sheet:

    org.postgresql.util.PSQLException: ERROR: duplicate key value violates unique constraint "survey_method_unique"
    Detail: Key (survey_id, method_id, block_num)=(923400005, 3, 1) already exists.

**Comment by utas-tajona** [comment]
Yes this is expected since we have missing validation related to methods and block

________________________________

**Comment by nspool** [comment]
Thanks, changing the date value fixes the constraint error. However am now getting a different internal server error:

    org.postgresql.util.PSQLException: ERROR: null value in column "observable_item_id" violates not-null constraint
    Detail: Failing row contains (2284994, 140047, 186, null, 52, 1, null).

Also these errors are not being displayed in the warning. The only thing showing in the webapp is "Internal Server Error".

**Comment by nspool** [comment]
Complete server log for `observable_item_id` submission error:
[backtrace.txt](https://github.com/aodn/nrmn-application/files/6354509/backtrace.txt)

**Comment by utas-tajona** [comment]
from the backtrace it seems to be the same error:   Detail: Key (survey_id, method_id, block_num)=(923400006, 2, 2) already exists.

**Comment by utas-tajona** [comment]
I feel  like I  should double check with @bpasquer  about this error.

I don't see any validation check left  about survey  block  methods. 
Maybe it's an issue where we have to retrieve this combo.

**Comment by ccmoloney** [comment]
Is this error saying that you have a duplicate Method 2/Block 2 in the survey you are ingesting?

**Comment by utas-tajona** [comment]
we are trying to add again the survey_method which shouldn't happen

**Comment by ccmoloney** [comment]
"I feel like I should double check with @bpasquer about this error.

I don't see any validation check left about survey block methods.
Maybe it's an issue where we have to retrieve this combo."

Are we missing a check @utas-tajona ?

**Comment by utas-tajona** [comment]
from https://github.com/aodn/backlog/issues/2933 and https://github.com/aodn/backlog/issues/2901 none of them as a check related to survey_id + method + block.

In the code we awlays are inserting the survey_id, method_id, block_id. which make sens. 
date / diver / site  + method + block should always be uniq

**Comment by utas-tajona** [comment]
so we are clearly missing a check if the survey_method already exists

**Comment by utas-tajona** [comment]
for the error message, the backendcurent implementation doesn't deal with error message. seems like another task to add

**Comment by nspool** [comment]
OK, will evaluate this PR just based on a successful condition.

**Comment by ccmoloney** [comment]
> from [aodn/backlog#2933](https://github.com/aodn/backlog/issues/2933) and [aodn/backlog#2901](https://github.com/aodn/backlog/issues/2901) none of them as a check related to survey_id + method + block.
> 
> In the code we awlays are inserting the survey_id, method_id, block_id. which make sens.
> date / diver / site + method + block should always be uniq

So: we are saying that we are not detecting duplicate blocks in the data for a new survey?

**Comment by utas-tajona** [comment]
re deployed: http://nrmn-fixing-ingest.dev.aodn.org.au/home

**Comment by nspool** [comment]
Created a new diver and site and made a new file with different dates. However still getting an error. File and logs from stack attached.

[TailLogs-1619063276401.txt](https://github.com/aodn/nrmn-application/files/6355379/TailLogs-1619063276401.txt)
[NATHAN TEST NEW SITE AND DIVER.xlsx](https://github.com/aodn/nrmn-application/files/6355380/NATHAN.TEST.NEW.SITE.AND.DIVER.xlsx)

**Comment by utas-tajona** [comment]
your sheet is full of duplicates 

ger | NEWSITE | New Site | -42.72013855 | 148.011459351 | 2/02/2018 | 7 | N | 11:00:00 | 0 | 5.2 | 1 | 0
-- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | --
ger | NEWSITE | New Site | -42.72013855 | 148.011459351 | 2/02/2018 | 7 | N | 11:00:00 | 0 | 5.1 | 1 | 0
ger | NEWSITE | New Site | -42.72013855 | 148.011459351 | 2/02/2018 | 7 | N | 11:00:00 | 0 | 5.2 | 1 | 0
ger | NEWSITE | New Site | -42.72013855 | 148.011459351 | 2/02/2018 | 7 | N | 11:00:00 | 0 | 5.1 | 1 | 0
ger | NEWSITE | New Site | -42.72013855 | 148.011459351 | 2/02/2018 | 7 | N | 11:00:00 | 0 | 5.2 | 1 | 0
ger | NEWSITE | New Site | -42.72013855 | 148.011459351 | 2/02/2018 | 7 | N | 11:00:00 | 0 | 5.1 | 1 | 0
ger | NEWSITE | New Site | -42.72013855 | 148.011459351 | 2/02/2018 | 7 | N | 11:00:00 | 0 | 5.2 | 1 | 0
ger | NEWSITE | New Site | -42.72013855 | 148.011459351 | 2/02/2018 | 7 | N | 11:00:00 | 0 | 5.2 | 1 | 0
ger | NEWSITE | New Site | -42.72013855 | 148.011459351 | 2/02/2018 | 7 | N | 11:00:00 | 0 | 5.4 | 1 | 0
ger | NEWSITE | New Site | -42.72013855 | 148.011459351 | 2/02/2018 | 7 | N | 11:00:00 | 0 | 5.3 | 1 | 0
ger | NEWSITE | New Site | -42.72013855 | 148.011459351 | 2/02/2018 | 7 | N | 11:00:00 | 0 | 5.4 | 1 | 0
ger | NEWSITE | New Site | -42.72013855 | 148.011459351 | 2/02/2018 | 7 | N | 11:00:00 | 0 | 5.3 | 1 | 0
ger | NEWSITE | New Site | -42.72013855 | 148.011459351 | 2/02/2018 | 7 | N | 11:00:00 | 0 | 5.4 | 1 | 0
ger | NEWSITE | New Site | -42.72013855 | 148.011459351 | 2/02/2018 | 7 | N | 11:00:00 | 0 | 5.4 | 1 | 0
ger | NEWSITE | New Site | -42.72013855 | 148.011459351 | 2/02/2018 | 7 | N | 11:00:00 | 0 | 5.4 | 1 | 0
ger | NEWSITE | New Site | -42.72013855 | 148.011459351 | 2/02/2018 | 7 | N | 11:00:00 | 0 | 5.3 | 1 | 0

get-newsite-2022018-1-0 is duplicated more than 10 times

**Comment by nspool** [comment]
@utas-tajona Tried again with a sheet with 3 lines but still seeing the internal server error. Which columns do I need to modify?

[NATHAN TEST NEW SITE AND DIVER SMALL.xlsx](https://github.com/aodn/nrmn-application/files/6355656/NATHAN.TEST.NEW.SITE.AND.DIVER.SMALL.xlsx)

**Comment by nspool** [comment]
Log for previous error: 
[catalina.txt](https://github.com/aodn/nrmn-application/files/6355721/catalina.txt)

**Comment by utas-tajona** [comment]
i managed to ingested it. Method 0 isn't supported in the current implementation. I will double check  with @bpasquer  when i can.

**Comment by ccmoloney** [comment]
I think this can be merged to allow testing by Bene in edge.

The focus of that testing should be to identify if the current set of validation rules are verifying what they should be & whether or not datasheet we are indicating is valid is, in fact, able to be ingested correctly.

**Comment by nspool** [comment]
@ccmoloney OK, will merge when checks are succeeding.

**Pull request merged**